ailab/db/finesse/prompt/qna_user_prompt.txt

```diff
@@ -1 +1 @@
-Your task is to process a JSON file containing text relevant to a search query. Produce a new JSON file completing all existing keys in the JSON template. Specifically, generate entries for 'question' and 'answer' where 'question' represents a question derived from the text and 'answer' provides an appropriate answer addressing the generated question.
\ No newline at end of file
+Your task is to process a JSON file containing text relevant to a search query. Produce a new JSON file completing all existing keys in the JSON template. Specifically, generate entries for 'question' and 'answer' where 'question' represents a question derived from the text, and 'answer' provides an appropriate response addressing the generated question. After that, determine a score into "question_quality" between 0.0 and 1.0 for the question you generated. The more complex the question is, the higher the score it receives. Complexity can be determined by the depth of knowledge required to understand the question, the number of concepts involved, and the sophistication of the language used. Every forms or PDF question like: "how do I fill a form?" or "how to open this document?" will be low scored.
```

Comment:
this makes complexity proportional to  question quality which are two different concepts. the main issue with "how do I fill a form?" is ... lack of context. which form?

"how to open this document?" - which document?

lack of context is what is causing a question to have to be excluded, as the question is not specific to be answered because it lacks context.

Also - evaluating quality should be separate from the generation step IMO, not tacked on with generation. We can also ask simpler yes or no question:

"does the question provide enough context to be answered as-is, yes or no?"

Reply:
I can do that, I'm not sure it will work because question sometimes are pretty clear and have context.

```
{
    "page_score": 0.48409016746766803,
    "question_quality": 0.2,
    "crawl_id": "",
    "chunk_id": "",
    "title": "Demande de rapport complet",
    "url": "https://inspection.canada.ca/salubrite-alimentaire-pour-l-industrie/chimie-et-microbiologie-alimentaires/bulletin-d-enquete-et-rapports-d-analyse-sur-la-sa/bacteries-pathogenes-dans-les-concombres/fra/1521093456907/1521093498650",
    "text_content": "Demande de rapport complet La version intégrale du présent rapport est disponible sur demande. Le format dans lequel vous souhaitez recevoir le rapport (obligatoire) Fichier électronique PDF (format de document portable) envoyé par courriel Fichier électronique RTF (Rich Text Format) envoyé par courriel Copie braille envoyée par la poste Copie papier envoyée par la poste Vos coordonnées Prénom : (obligatoire) Nom : (obligatoire) Adresse de courriel : (obligatoire) Confirmer Adresse de courriel : (obligatoire) Comment vous vous décriviez? (obligatoire) Académie Association sectorielle Autres institutions du gouvernement Consommateur / Grand public Groupe de consommateur Industrie / Secteur commercial Adresse 1 : Adresse 2 : Ville : Province : Pays : Code postal :",
    "question": "Comment puis-je demander un rapport complet?",
    "answer": "Pour demander un rapport complet, vous devez fournir vos coordonnées, y compris votre prénom, nom, adresse e-mail et adresse postale. Vous devez également spécifier le format dans lequel vous souhaitez recevoir le rapport, tel que PDF électronique ou copie papier. Une fois que vous avez fourni toutes les informations requises, vous pouvez soumettre votre demande."
}
```

So I'll try to have a boolean telling us if there is enough context in the **question**. Do you think we should do that in another issue tho? @rngadam